src/unity/python/turicreate/toolkits/image_classifier/image_classifier.py

```diff
@@ -545,12 +551,163 @@ def evaluate(self, dataset, metric='auto', verbose=True, batch_size=64):
           >>> results = model.evaluate(data)
           >>> print results['accuracy']
         """
+
+        import os, json, math
+
         if(batch_size < 1):
             raise ValueError("'batch_size' must be greater than or equal to 1")
 
         extracted_features = self._extract_features(dataset, verbose=verbose, batch_size=batch_size)
         extracted_features[self.target] = dataset[self.target]
-        return self.classifier.evaluate(extracted_features, metric = metric)
+        
+        metrics = self.classifier.evaluate(extracted_features, metric = metric)
+        predictions = metrics["predictions"]["probs"]
+        state = self.__proxy__.get_state()
+        labels = state["classes"]
+
+
+        def entropy(probs):
+            if _sys.version_info >= (3, 0):
```

Comment:
You can use [six](https://pythonhosted.org/six/index.html?highlight=reduce#module-six.moves) to get a version-independent `reduce`. We already have a dependency on it. I think you would do:

```
from six.moves import reduce
```

And it should work on both Python versions (no need for functools or built-in).